homeassistant/components/home_connect/sensor.py

```diff
@@ -38,47 +41,183 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+EVENT_OPTIONS = ["confirmed", "off", "present"]
+
+
 @dataclass(frozen=True, kw_only=True)
 class HomeConnectSensorEntityDescription(SensorEntityDescription):
     """Entity Description class for sensors."""
 
-    device_class: SensorDeviceClass | None = SensorDeviceClass.ENUM
-    options: list[str] | None = field(
-        default_factory=lambda: ["confirmed", "off", "present"]
-    )
-    desc: str
-    appliance_types: tuple[str, ...]
+    default_value: str | None = None
+    appliance_types: tuple[str, ...] | None = None
+    sign: int = 1
+
+
+BSH_PROGRAM_SENSORS = (
+    HomeConnectSensorEntityDescription(
+        key="BSH.Common.Option.RemainingProgramTime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        sign=1,
+        translation_key="remaining_program_time",
+    ),
+    HomeConnectSensorEntityDescription(
+        key="BSH.Common.Option.Duration",
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        sign=1,
+        translation_key="duration",
```

Comment:
```suggestion
```
This one can be removed as if there is no translation key and a device class, it will take the device class name